consensus/packets/phase1packet.go

```diff
@@ -39,27 +40,63 @@ type Phase1Packet struct {
 	Signature [SignatureLength]byte
 }
 
-func NewPhase1Packet() *Phase1Packet {
-	return &Phase1Packet{}
+func (p1p *Phase1Packet) GetOrigin() core.ShortNodeID {
+	return p1p.packetHeader.OriginNodeID
 }
 
-func (p1p *Phase1Packet) hasPulseDataExt() bool { // nolint: megacheck
-	return p1p.packetHeader.f00
+func (p1p *Phase1Packet) GetTarget() core.ShortNodeID {
+	return p1p.packetHeader.TargetNodeID
 }
 
-func (p1p *Phase1Packet) hasSection2() bool {
-	return p1p.packetHeader.f01
+func (p1p *Phase1Packet) SetRouting(origin, target core.ShortNodeID) {
+	p1p.packetHeader.OriginNodeID = origin
+	p1p.packetHeader.TargetNodeID = target
+	p1p.packetHeader.HasRouting = true
+}
+
+func (p1p *Phase1Packet) GetType() PacketType {
+	return p1p.packetHeader.PacketT
 }
 
-func (p1p *Phase1Packet) SetPacketHeader(header *RoutingHeader) error {
-	if header.PacketType != types.Phase1 {
-		return errors.New("Phase1Packet.SetPacketHeader: wrong packet type")
+func (p1p *Phase1Packet) Verify(crypto core.CryptographyService, key crypto.PublicKey) error {
+	raw, err := p1p.rawBytes()
+	if err != nil {
+		return errors.Wrap(err, "Failed to get raw part of phase 1 packet")
 	}
-	p1p.packetHeader.setRoutingFields(header, Phase1)
+	valid := crypto.Verify(key, core.SignatureFromBytes(p1p.Signature[:]), raw)
+	if !valid {
+		return errors.New("bad signature")
+	}
+	return nil
+}
 
+func (p1p *Phase1Packet) Sign(crypto core.CryptographyService) error {
```

Comment:
Use `cryptographyService` name for `core.CryptographyService ` otherwise it's highly unclear and shadowing `crypto` package.